docs/proposals/config_data.md

```diff
@@ -0,0 +1,319 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores data used for the configuration of ``Kubernetes`` components.
+
+The main focus points of this proposal are:
+
+* Solving dynamic configuration problem of components.
+* Encapsulate configuration information to decrease component configuration complexity.
+* Create a flexible configuration model for ``Kubernetes`` components.
+* Increase overall system quality.
+
+## Motivation
+
+Currently command-line flags and environment variables are used for component configurations in ``Kubernetes``
+which is complicated and error-prone. This approach is not suitable for a distributed computing environment.
+Specifically the problems with the current approach are:
+
+* Complicated deployment orchestration
+* Hard to synchronize across replicated components.
+* Hard to change dynamically as needed by continuous services.
+* Hard to change in different versions that may support different configuration options.
+
+## Solution
+
+A new ``ConfigData`` resource is proposed to solve the aforementioned problems
+that is stored in ``etcd`` and distributed to interested/necessary components
+through the API server. 
+
+The ``ConfigData`` architecture will be very similar to that of ``Secrets``.
```

Comment:
I'm using double backticks to indicate that this is an object in the codebase. It becomes hard to notice small errors if these are not distinguished from the other parts. It's not distracting I hope?